internal/controller/replication.storage/volumereplication_controller.go

```diff
@@ -113,7 +111,7 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	vrcObj, err := r.getVolumeReplicationClass(logger, instance.Spec.VolumeReplicationClass)
 	if err != nil {
 		setFailureCondition(instance, "failed to get volumeReplication class", err.Error(), instance.Spec.DataSource.Kind)
-		uErr := r.updateReplicationStatus(instance, logger, getCurrentReplicationState(instance), err.Error())
+		uErr := r.updateReplicationStatus(instance, logger, GetCurrentReplicationState(instance.Status.State), err.Error())
```

Comment:
changes like this could have been a separate commit

Reply:
Ack, moved it to a separate commit.